src/bit_machine/mod.rs

```diff
@@ -365,67 +369,74 @@ impl BitMachine {
         }
     }
 
-    fn exec_jet<J: Jet>(&mut self, jet: J, env: &J::Environment) -> Result<(), JetFailed> {
-        use simplicity_sys::c_jets::frame_ffi::{c_readBit, c_writeBit, CFrameItem};
-        use simplicity_sys::c_jets::uword_width;
-        use simplicity_sys::ffi::UWORD;
+    /// Return a read frame that is equivalent to the active read frame.
+    fn get_input_frame(&mut self, bit_width: usize, buffer: &mut Vec<UWORD>) -> CFrameItem {
+        let buffer_width = uword_width(bit_width);
+        buffer.clear();
+        buffer.resize(buffer_width, 0);
```

Comment:
In 63c2d233fb29a8837e930d79507d89d06a4443d9:

I would instead accept a `&mut [u8]` and `assert_eq` that its length is correct, and make the caller do the resizing and initialization. Then at least we're not invalidating other `CFrameItem`s that refer to this object from within this function.

Reply:
I will look into this tomorrow